### PR TITLE
add note about binary data storage

### PR DESCRIPTION
**Pull request by StarfallProjects**

### CODE REVIEW

**Comment by StarfallProjects** [comment]
In response to: https://github.com/n8n-io/n8n/issues/5138

File: docs/hosting/scaling/execution-modes-processes.md [hunk 1]
```diff
@@ -1,38 +1,38 @@
...
+Set the n8n mode using the following [environment variables](/hosting/environment-variables/environment-variables/#executions):
 
-| Variable | Options | Description |
-| -------- | ------- | ----------- |
-| *EXECUTIONS_PROCESS* | `own` (default), `main`  | Determines whether each execution runs in its own process, or all run in the  main process. |
-| *EXECUTIONS_MODE* | `regular` (default), `queue` | Determines whether executions are processed by a single instance or are distributed across worker instances. |
+| Variable | Options | Default | Description |
+| :------- | :------ | :------ | :--------- |
+| `EXECUTIONS_PROCESS` | Enum string: `main`, `own` | `own` | Whether n8n executions run in their own process or the main process. |
+| `EXECUTIONS_MODE` | Enum string: `regular`, `queue` | `regular` | Whether executions should run directly or using queue. |
```
Comment (krynble) [inline docs/hosting/scaling/execution-modes-processes.md:10]: One thing to note is that when `EXECUTIONS_MODE` is set to queue, the `EXECUTIONS_PROCESS` is ignored. n8n will behave as if you're using `main` in the worker instances.

File: docs/hosting/scaling/execution-modes-processes.md [hunk 2]
```diff
@@ -45,8 +45,11 @@ This is the default setting for n8n. When running n8n in the `regular` mode, eve
 
 ### Queue
 
-`Queue` mode is designed for handling very high workloads. In this mode you run multiple instances of n8n: one ‘main' instance coordinates, and other ‘worker' instances actually process the executions. You can also add dedicated instances to handle incoming webhooks.
+`Queue` mode is designed for handling high workloads. In this mode you run multiple instances of n8n: one main instance coordinates work, and other worker instances process the executions. You can also add dedicated instances to handle incoming webhooks.
```
Comment (krynble) [inline docs/hosting/scaling/execution-modes-processes.md:48]:
```suggestion
`Queue` mode is designed for handling high workloads. In this mode you run multiple instances of n8n: one main instance is required for triggering some workflows and allowing user access via UI, while the other worker instances process the executions. You can also add dedicated instances to handle incoming webhooks.
```

File: docs/hosting/scaling/execution-modes-processes.md [hunk 3]
```diff
@@ -45,8 +45,11 @@ This is the default setting for n8n. When running n8n in the `regular` mode, eve
...
 
 `Queue` mode requires more setup than `regular` mode, but provides great scalability and stability with minimal latency.
 
+!!! note "Binary data storage"
+	n8n doesn't support queue mode with binary data storage. If your workflows need to persist binary data, you can't use queue mode.
```
Comment (krynble) [inline docs/hosting/scaling/execution-modes-processes.md:53]: This made me realize that we don't seem to have a dedicated documentation for binary data storage.

Basically, when handling what we call "binary data" (files) that can be large, this data stays in memory and takes up a lot of space.

For simply moving files (download and upload, for instance), this is unnecessary and could cause n8n to crash.

Binary data storage has 2 options: in memory (default) or filesystem (where n8n saves data to disk instead, saving memory space).

I think we need to add this to our docs somewhere.

image

The settings for this feature can be seen here

**Review by krynble — CHANGES_REQUESTED** [review]

File: docs/hosting/scaling/execution-modes-processes.md (hunk 3 above)
Comment (StarfallProjects) [reply docs/hosting/scaling/execution-modes-processes.md:53]: Added a doc about binary data, and split out the env vars in the env vars doc to make them more discoverable

**Review by krynble — APPROVED** [review]

**Pull request merged**